Provider/ImageProvider.php

```diff
@@ -72,12 +72,32 @@ public function getHelperProperties(MediaInterface $media, $format, $options = a
             $box = $this->resizer->getBox($media, $resizerFormat);
         }
 
+        $mediaWidth = $box->getWidth();
+
+        $srcset = array();
+        $context = $media->getContext();
+
+        foreach ($this->getFormats() as $providerFormat => $settings) {
+            // Check if format belongs to the current media's context
+            if (substr($providerFormat, 0, strlen($context)) == $context) {
```

Comment:
Maybe we can leave it like that for now for consistency and add a $media->hasFormat() function using strpos in a different pull request and update the different places where it is used?